RealifeTech-CoreSDK/APILayer/Tests/DiskCacheTests.swift

```diff
@@ -9,61 +9,163 @@
 import XCTest
 @testable import RealifeTech_CoreSDK
 
-struct TestObj: Codable {
-    let id: String
-    let otherField: Int
-}
+final class DiskCacheTests: XCTestCase {
 
-struct TestObjLocalRepository: LocalDiskCacheDataProviding {
-    typealias Cdble = TestObj
-}
+    private var fileManager: MockFileManager!
+    private var clearCacheQueue: DispatchQueue!
+    private var sut: DiskCache!
+
+    private let file = "to_save_content"
+    private let fileName = "test_file"
+    private let privateFileName = "private_test_file" + DiskCache.privateIndicator
+    private let expiresAt: Int64 = Date().toMilliseconds() + 3000
+
+    private var expectedSavedPath: URL {
+        fileManager.temporaryDirectory.appendingPathComponent("\(fileName)\(DiskCache.fileExtension)")
+    }
+    private var expectedSavedPrivateObjectPath: URL {
+        fileManager.temporaryDirectory.appendingPathComponent("\(privateFileName)\(DiskCache.fileExtension)")
+    }
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        fileManager = MockFileManager()
+        clearCacheQueue = DispatchQueue(label: "DiskCacheTests Clear Cache Queue")
+        sut = DiskCache(fileManager: fileManager, clearCacheQueue: clearCacheQueue)
+    }
+
+    override func tearDownWithError() throws {
+        if fileManager.fileExists(atPath: expectedSavedPath.path) {
+            try fileManager.removeItem(at: expectedSavedPath)
+        }
+        if fileManager.fileExists(atPath: expectedSavedPath.path) {
```

Comment:
This is duplicated from the lines above. Can it be removed?

Reply:
Good catch. This should be `expectedSavedPrivateObjectPath`